### PR TITLE
[Merged by Bors] - Remove Pyrmont testnet

### PR DESCRIPTION
**Pull request by tqpcharlie**
## Issue Addressed

N/A

## Proposed Changes

Removes all configurations and hard-coded rules related to the deprecated Pyrmont testnet.

## Additional Info

Pyrmont is deprecated/will be shut down after being used for scenario testing, this PR removes configurations related to it.

### CODE REVIEW

**Comment by tqpcharlie** [comment]
This is currently a draft because Pyrmont hasn't been shut down yet, this is just preparation for when it will.

**Comment by paulhauner** [comment]
I'm going to close this since it's not clear when we should remove the flag and it's nice to keep the open PRs clean.

This PR will remain and we can re-open when it's time to remove Pyrmont.

**Comment by paulhauner** [comment]
After a suggestion from @macladson, I'm tempted to remove Pyrmont now. I've posted on [Discord](https://discord.com/channels/595666850260713488/595701053161209866/940128998137139220) to see if anyone objects.

@Zachinquarantine are you interested in fixing the conflicts? No pressure, we can manage otherwise.

**Comment by tqpcharlie** [comment]
> After a suggestion from @macladson, I'm tempted to remove Pyrmont now. I've posted on [Discord](https://discord.com/channels/595666850260713488/595701053161209866/940128998137139220) to see if anyone objects.
> 
> @Zachinquarantine are you interested in fixing the conflicts? No pressure, we can manage otherwise.

will do!

**Comment by tqpcharlie** [comment]
Done, ready!

File: common/eth2_network_config/src/lib.rs
Comment (macladson) [inline common/eth2_network_config/src/lib.rs]: Looks like there's some formatting issues (I suspect it's due to there not being a newline here)
Would you mind getting `cargo fmt` passing so that the rest of CI can run?

**Review by macladson — CHANGES_REQUESTED** [review]
Thanks for doing this! 
It looks good so far, but it seems like there a few other instances of `pyrmont` still in the code base.
I _think_ all of these can be removed or changed to reference `prater` (in the case of documentation).

The files that I found with references to `pyrmont` were:
- `testing/web3signer_tests/src/lib.rs`
- `common/eth2_network_config/built_in_network_configs/pyrmont/config.yaml`
- `book/src/wallet-create.md`
- `book/src/validator-create.md`
- `book/src/mainnet-validator.md`
- `book/src/key-management.md`
- `book/src/faq.md`
- `book/src/docker.md`
- `.github/workflows/release.yml`

**Review by macladson — APPROVED** [review]
Nice work :slightly_smiling_face: 
This looks good from my perspective!
I'll let a maintainer give this a once-over, then we can merge.

**Review by michaelsproul — APPROVED** [review]
Awesome! Thank you!

bors r+

**Pull request closed without merging**
